Zemismart/__init__.py

```diff
@@ -111,17 +157,47 @@ def handleNotification(self, handle, data):
             elif data[1] == self.get_position_cmd[0]:
                 pos = data[5]
                 self.save_position(pos)
-                self.last_command_status = True
-            elif data[1] in (self.finished_moving_cmd[0], self.notify_position_cmd[0]):
+                self._get_position_responses.add(data[1])
+                self._check_get_position_response()
+            elif data[1] == self.finished_moving_cmd[0]:
                 pos = data[4]
                 self.save_position(pos)
                 self.last_command_status = True
-            elif data[1] in (self.set_position_cmd[0], self.pin_cmd[0], self.move_cmd[0]):
+            elif data[1] == self.unknown_cmd_a9[0]:
+                self._get_position_responses.add(data[1])
+                self._check_get_position_response()
+            elif data[1] == self.get_timers_cmd[0]:
+                self._get_position_responses.add(data[1])
+                self.timers = []
+                # Single timer takes 5 bytes
+                for offset in range(0, int(int(data[2]) / 5)):
+                    timer_data_start = 3 + offset * 5
+                    timer = Zemismart.Timer(
+                        bool(data[timer_data_start]),
+                        int(data[timer_data_start + 1]),
+                        int(data[timer_data_start + 2]),
+                        int(data[timer_data_start + 3]),
+                        int(data[timer_data_start + 4])
+                    )
+                    self.timers.append(timer)
+
+                self._check_get_position_response()

```

Comment:
Why use `_check_get_position_response()` when this has nothing to do with the get position command? Also, why add this command to `_get_position_responses` on line 167?

Reply:
Technically, there's no separate command for "get timers". When you send command 0xA7 ("get position") to an AM43 device, the device always responds with 3 separate messages: 0xA7, 0xA8, and 0xA9.
I didn't come up with a better name for a method that will be checking whether all the messages related to "get position" are received.

Reply:
You are right. Maybe 0xA7 is not a "get position" command, but a "get status" command or similar. I think this is ok, for now, but the handling of responses can probably be improved a lot.

Reply:
Good point, I'll change "get position" to "get status".